src/CP/constraints/absolute.jl

```diff
@@ -26,15 +26,24 @@ end
 function propagate!(constraint::Absolute, toPropagate::Set{Constraint}, prunedDomains::CPModification)
 
     if isbound(constraint.x) 
-        prunedY = assign!(constraint.y, abs(assignedValue(constraint.x)))
+        value = abs(assignedValue(constraint.x))
+        if !(value in constraint.y.domain)
+            return false
+        end
+
+        prunedY = assign!(constraint.y, value)
         addToPrunedDomains!(prunedDomains, constraint.y, prunedY)
         triggerDomainChange!(toPropagate, constraint.y)
         setValue!(constraint.active, false)
 
     elseif isbound(constraint.y) # x is equal to y or -y
+        value = assignedValue(constraint.y)
+        if !(value in constraint.x.domain || -value in constraint.x.domain)
+            return false
+        end
     
         for v in minimum(constraint.x.domain):maximum(constraint.x.domain)
-            if v != assignedValue(constraint.y) && v != -assignedValue(constraint.y)
+            if v != value && v != -value
```

Comment:
On second thought, this case isn't necessary as it doesn't trigger errors and will be handled at the end if `x.domain` becomes empty. Should we keep it for symmetry though ?